framework/src/modules/chain/helpers/jobs_queue.js

```diff
@@ -51,17 +51,29 @@ const jobsQueue = {
 		// Check if job is function, name is string and time is integer
 		if (
 			!job ||
-			Object.prototype.toString.call(job) !== '[object Function]' ||
+			!(
+				Object.prototype.toString.call(job) === '[object Function]' ||
+				Object.prototype.toString.call(job) === '[object AsyncFunction]'
+			) ||
+			job.length > 1 ||
 			typeof name !== 'string' ||
 			!Number.isInteger(time)
 		) {
 			throw new Error('Syntax error - invalid parameters supplied');
 		}
 
 		const nextJob = function() {
-			return job(() => {
+			const nextJobStep = () => {
 				jobsQueue.jobs[name] = setTimeout(nextJob, time);
-			});
+			};
+
+			// Job was called with a callback
+			if (Object.prototype.toString.call(job) === '[object Function]') {
```

Comment:
I think I've taken into account all the possibilities in the if control. Check again please.

Reply:
Its not necessary that job must be an async function constructor. It can be normal function returning the promise. So we just check if function takes an argument that as a callback, else it must return a promise, no matter its an async function or not.